packages/myst-cli/src/build/build.ts

```diff
@@ -54,14 +55,33 @@ export async function collectAllBuildExportOptions(
   files: string[],
   opts: BuildOpts,
 ) {
-  const { force } = opts;
-  const formats = getExportFormats({ ...opts, explicit: files.length > 0 });
+  const { force, output } = opts;
+  if (output && files.length !== 1) {
+    throw new Error('When specifying a named output for export, you must list exactly one file.');
+  }
+  const formats = getExportFormats({
+    ...opts,
+    explicit: files.length > 0,
+    extension: output ? path.extname(output) : undefined,
+  });
+  if (output && formats.length !== 1) {
+    throw new Error(`Unrecognized file extension for output: ${path.extname(output)}`);
+  }
   session.log.debug(`Exporting formats: "${formats.join('", "')}"`);
   let exportOptionsList: ExportWithInputOutput[];
   if (files.length) {
     exportOptionsList = await collectExportOptions(session, files, formats, {
-      force,
+      // If there is an output and file specified, force is implied
+      force: force || !!output,
```

Comment:
Do we also want implied `force` if, like `(files.length === 1 && (opts.docx || opts.pdf || opts.tex || opts.xml))`? I think that would be the case for `myst build index.md --jats`

Reply:
Updated!